crates/lang/ir/src/ir/item_impl/constructor.rs

```diff
@@ -64,6 +64,8 @@ use syn::spanned::Spanned as _;
 pub struct Constructor {
     /// The underlying Rust method item.
     pub(super) item: syn::ImplItemMethod,
+    /// If the ink! constructor is a wildcard selector.
+    has_wildcard_selector: bool,
```

Comment:
Same as for ink! messages: I think it would be cleaner to use `selector: Option<SelectorOrWildcard>` here instead of both current `has_wildcard_selector` and `selector` fields since they can represent invalid state (`(true, Some(_))`) which my proposed field cannot.